share/spack/setup-env.sh

```diff
@@ -189,7 +189,23 @@ _sp_prefix=$(cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
 _spack_pathadd PATH       "${_sp_prefix%/}/bin"
 
 _sp_sys_type=$(spack-python -c 'print(spack.architecture.sys_type())')
-_sp_dotkit_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('dotkit')))")
-_sp_tcl_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('tcl')))")
+
+_sp_use_custom_moduleroots="$(grep module_roots "${_sp_prefix}/etc/spack/config.yaml" > /dev/null 2>&1)"
```

Comment:
Project-level config scopes (on the command line) are not yet a supported feature.  If/when they are, I suppose we would have to specify them also on the command line to `setup-env.sh`?  That would be a bummer...

But I don't understand the issue very well at this point.